src/parser/mod.rs

```diff
@@ -475,6 +475,10 @@ impl<'a> Parser<'a> {
                     if expecting_statement_delimiter && word.keyword == Keyword::END {
                         break;
                     }
+
```

Comment:
@aharpervc Sorry for the delay getting back to this, actually I'm leaning towards not merging this feature after all unfortunately. It feels to me like there's a bit too many special cases (like this extending the core parsing loop for example) for this feature that go against the normal of sql statement parsing. And combined with this feature not being part of any sql reference but rather a specific tool, it makes the tradeoff undesirable